tests/imap_integration.rs

```diff
@@ -670,3 +699,83 @@ fn qresync() {
     // Assert that the new highest mod sequence is being returned
     assert_ne!(r.mod_seq, None);
 }
+
+fn assert_quota_resource(
+    resource: &imap::types::QuotaResource,
+    name: imap::types::QuotaResourceName,
+    limit: u64,
+    usage: Option<u64>,
+) {
+    assert_eq!(resource.name, name);
+    if let Some(usage) = usage {
+        assert_eq!(resource.usage, usage);
+    }
+    assert_eq!(resource.limit, limit);
+}
+
+#[test]
+fn quota() {
+    use imap::types::{QuotaResourceLimit, QuotaResourceName};
+
+    let greeting = get_greeting();
+    let is_greenmail = greeting.find("Cyrus").is_none();
+
+    let to = "inbox-quota@localhost";
+
+    if is_greenmail {
+        let mut c = session(to);
+
+        let quota_root = c.get_quota_root("INBOX").unwrap();
+
+        assert_eq!(quota_root.mailbox_name(), "INBOX");
+
+        let root_names = quota_root.quota_root_names().collect::<Vec<&str>>();
+        assert_eq!(root_names, Vec::<&str>::new());
+        // TODO build tests for greenmail
+    } else {
+        // because we are cyrus we can "test" the admin account for checking the GET/SET commands
+        let mut admin = session_with_options("cyrus", false);
```

Comment:
Could you add a comment about why we specifically need to _not_ `clean_mailbox` for the admin session?